cns/restserver/ipam.go

```diff
@@ -300,15 +323,18 @@ func (service *HTTPRestService) releaseIPConfig(podInfo cns.KubernetesPodInfo) e
 	ipID := service.PodIPIDByOrchestratorContext[podInfo.GetOrchestratorContextKey()]
 	if ipID != "" {
 		if ipconfig, isExist := service.PodIPConfigState[ipID]; isExist {
-			service.setIPConfigAsAvailable(ipconfig, podInfo)
-			logger.Printf("Released IP %+v for pod %+v", ipconfig.IPAddress, podInfo)
+			_, err := service.setIPConfigAsAvailable(ipconfig, podInfo)
+			if err != nil {
+				return fmt.Errorf("[releaseIPConfig] failed to mark IPConfig [%+v] as Available.", ipconfig.ToString(), err)
+			}
+			logger.Printf("[releaseIPConfig] Released IP %+v for pod %+v", ipconfig.IPAddress, podInfo)
 
 		} else {
-			logger.Errorf("Failed to get release ipconfig. Pod to IPID exists, but IPID to IPConfig doesn't exist, CNS State potentially corrupt")
-			return fmt.Errorf("releaseIPConfig failed. Pod to IPID exists, but IPID to IPConfig doesn't exist, CNS State potentially corrupt")
+			logger.Errorf("[releaseIPConfig] Failed to get release ipconfig. Pod to IPID exists, but IPID to IPConfig doesn't exist, CNS State potentially corrupt")
+			return fmt.Errorf("[releaseIPConfig] releaseIPConfig failed. Pod to IPID exists, but IPID to IPConfig doesn't exist, CNS State potentially corrupt")
 		}
 	} else {
-		logger.Errorf("SetIPConfigAsAvailable failed to release, no allocation found for pod")
+		logger.Errorf("[releaseIPConfig] SetIPConfigAsAvailable failed to release, no allocation found for pod")
```

Comment:
Should more context be added to these errors and logs? Without them, they are hard to correlate to the overall request

Reply:
This was getting covered in Shufang's PR hence didnt change it. But since I'm touching it anyways, will fix it

---
In reply to: [585065690](https://github.com/Azure/azure-container-networking/pull/806#discussion_r585065690) [](ancestors = 585065690)